.github/workflows/run-tests.yml

```diff
@@ -0,0 +1,76 @@
+name: Keycloak Client Tests
+
+on:
+  workflow_call:
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  client-tests:
+    name: Client tests
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    strategy:
+      matrix:
+        keycloak_server_version: [ "24.0", "26.0", "26.1", "nightly" ]
+    steps:
+      - uses: actions/checkout@v4
+
+      - id: test-setup
+        name: Test setup
+        uses: ./.github/actions/test-setup
+
+      - name: Run client tests
+        run: |
+          mvn -B -f testsuite/providers/pom.xml package -DskipTests=true
+          mvn -B -f testsuite/admin-client-tests/pom.xml test -Dkeycloak.version.docker.image=${{ matrix.keycloak_server_version }}
+          mvn -B -f testsuite/authz-tests/pom.xml test -Dkeycloak.version.docker.image=${{ matrix.keycloak_server_version }}
+
+  client-tests-java-11:
+    name: Client tests (Java 11)
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    strategy:
+      matrix:
+        keycloak_server_version: [ "nightly" ]
```

Comment:
You mentioned in your comment only the nightly is tested for Java 11, is there a particular reason? Or is it just to prevent a massive test execution on CI?

Reply:
No, no, we can test anything, this was just to setup something. We can do whatever you prefer. But I don't know if we want to test all again.

Reply:
+1 to test just `nightly` with Java 11. IMO sufficient as a compromise, so we have a test coverage for Java 11, but at the same time not need to test whole matrix with Java 11 to save some resources etc.